core/client/fs/src/main/java/alluxio/client/block/stream/NettyPacketWriter.java

```diff
@@ -256,6 +268,14 @@ public void close() throws IOException {
       if (mChannel.isOpen()) {
         mChannel.pipeline().removeLast();
       }
+      if (closeFuture != null) {
```

Comment:
I thought it is safe to call mChannel.pipeline().removeLast() after the channel is closed. I realized it is actually not after testing it. So you are right, i have to call closeFuture.sync() before removeLast().